# Notothenioid Hemoglobin cluster repeat annotations

Script for annotating and processing repeat annotations in the hemoblogin clusters of several notothenioid fish species and outgroups.

## Running Repeat Masker

Running `RepeatMasker` using the script `run_repeat_masker.sh`. It uses the repeat annotation generated by [Rivera-Colon et al. 2023 *MBE*](https://doi.org/10.1093/molbev/msad029). 

It takes the FASTA for the target genome and the FASTA for the repeat annotation library.

## Run the repeat landscape

Process the `RepeatMasker` output to generate the DIVSUM file needed for the repeat landscape plots. First, calculate the Kimura distance using the `RepeatMasker` alignments (`*.align`) using the `RepeatMaker` utility script `calcDivergenceFromAlign.pl`. This generates a based DIVSUM file, which is processed and tabulated to generate clearner, more descriptive outputs (see `parse_repeat_masker_out.py`).

## Parse the Repeat Masker output

Custom script used for processing and cleaning the `RepeatMasker` outputs. Takes the cross match file (`*.out`) from `RepeatMasker` along with the DIVSUM table from the utility scripts and generates a single table with the repeat annotation coordinates and the Kimura distance, all in a single file.

### Usage

```sh
parse_repeat_masker_out.py started on 2024-04-16 15:07:27

usage: parse_repeat_masker_out.py [-h] -c CROSS_MATCH -d DIVSUM [-o OUTDIR]
                                  [-m MIN_LENGTH]

Parse and merge the output from RepeatMasker.

optional arguments:
  -h, --help            show this help message and exit
  -c CROSS_MATCH, --cross-match CROSS_MATCH
                        (str) Path to the cross match (*.out) table from
                        Repeat Masker.
  -d DIVSUM, --divsum DIVSUM
                        (str) Path to the divergence summary (*.divsum) table
                        from Repeat Masker.
  -o OUTDIR, --outdir OUTDIR
                        (str) Path to output directory.
  -m MIN_LENGTH, --min-length MIN_LENGTH
                        (int) Minimum length of well characterized bases
                        (wellCharLen) needed to keep a sequence [default=10]
```

### Output table

The output is `repeat_masked_merged.tsv`.

```sh
#Name              Class  Family          Chromosome  StartBP   EndBP     WellCharLen  Kimura  SwScore
DR0015272          DNA    DNA/hAT-Tip100  2           21376111  21376206  2871         0.2092  428
DR0103000          LINE   LINE/Penelope   5           11192568  11192678  1362         0.2191  287
rnd-6_family-6844  LINE   LINE/L1         6           17114676  17114930  20683        0.0678  1836
rnd-1_family-857   DNA    DNA/hAT-Ac      11          477380    477612    52874        0.0226  1827
DR0081550          DNA    DNA/hAT-Ac      13          13640239  13640312  29984        0.0661  586
DR0081469          LINE   LINE/L2         16          32010056  32010142  176647       0.2081  328
rnd-1_family-249   LINE   LINE/RTE-BovB   16          9067434   9068658   1842036      0.0158  10179
DR0140129          SINE   SINE/MIR        20          13676319  13676332  5105         0.2455  262
rnd-1_family-656   DNA    DNA/TcMar-Tc2   23          44569505  44569794  572460       0.031   1942
DR0181611          DNA    DNA/TcMar-Tc1   24          13315996  13316193  89004        0.0283  1629
```

## Plot repeat landscape

Plot the final repeat landscape from the tabulated DIVSUM file (`*.divsum.tsv`) using `ggplot2`. This scripts looks over all target taxa/HB clusters and makes all plots with the same parameters.

## Author

Angel G. Rivera-Colon  
Institute of Ecology and Evolution  
University of Oregon

