# Notothenioid Hemoglobin cluster repeat annotations

Script for annotating and processing repeat annotations in the hemoblogin clusters of several notothenioid fish species and outgroups.

## Running Repeat Masker

Running `RepeatMasker` using the script `run_repeat_masker.sh`. It uses the repeat annotation generated by [Rivera-Colon et al. 2023 *MBE*](https://doi.org/10.1093/molbev/msad029). 

It takes the FASTA for the target genome and the FASTA for the repeat annotation library.

## Run the repeat landscape

Process the `RepeatMasker` output to generate the DIVSUM file needed for the repeat landscape plots. First, calculate the Kimura distance using the `RepeatMasker` alignments (`*.align`) using the `RepeatMaker` utility script `calcDivergenceFromAlign.pl`. This generates a based DIVSUM file, which is processed and tabulated to generate clearner, more descriptive outputs (see `parse_repeat_masker_out.py`).

## Parse the Repeat Masker output

Custom script used for processing and cleaning the `RepeatMasker` outputs. Takes the cross match file (`*.out`) from `RepeatMasker` along with the DIVSUM table from the utility scripts and generates a single table with the repeat annotation coordinates and the Kimura distance, all in a single file.

### Usage

```sh
parse_repeat_masker_out.py started on 2024-04-16 15:07:27

usage: parse_repeat_masker_out.py [-h] -c CROSS_MATCH -d DIVSUM [-o OUTDIR]
                                  [-m MIN_LENGTH]

Parse and merge the output from RepeatMasker.

optional arguments:
  -h, --help            show this help message and exit
  -c CROSS_MATCH, --cross-match CROSS_MATCH
                        (str) Path to the cross match (*.out) table from
                        Repeat Masker.
  -d DIVSUM, --divsum DIVSUM
                        (str) Path to the divergence summary (*.divsum) table
                        from Repeat Masker.
  -o OUTDIR, --outdir OUTDIR
                        (str) Path to output directory.
  -m MIN_LENGTH, --min-length MIN_LENGTH
                        (int) Minimum length of well characterized bases
                        (wellCharLen) needed to keep a sequence [default=10]
```

### Output table

The output is `repeat_masked_merged.tsv`.

```sh
#Chromosome  StartBP  EndBP  Strand  Name              Class          Family          WellCharLen  Kimura  SwScore
Chr1         198      233    -       DR0113934         LTR            LTR/DIRS        36           0.1587  227
Chr1         787      1254   -       DR0082056         DNA            DNA/hAT-Tip100  449          0.1055  2701
Chr1         1252     4453   +       rnd-6_family-740  LINE           LINE/L1-Tx1     3182         0.0314  26037
Chr1         4412     4488   -       DR0152715         Unknown        Unknown         34           0.1485  229
Chr1         4499     4654   +       DR0081825         Unknown        Unknown         149          0.1097  2829
Chr1         4655     5423   -       DR0081478         DNA            DNA/hAT-Tip100  759          0.0285  6047
Chr1         6017     6043   +       (ATTT)n           Simple_repeat  Simple_repeat   None         None    28
Chr1         6765     6856   -       DR0115825         DNA            DNA/hAT-Ac      87           0.2788  241
Chr1         7000     7126   +       DR0081603         DNA            DNA/hAT-hAT5    82           0.0505  613
```

## Plot repeat landscape

Plot the final repeat landscape from the tabulated DIVSUM file (`*.divsum.tsv`) using `ggplot2`. This scripts looks over all target taxa/HB clusters and makes all plots with the same parameters.

## Author

Angel G. Rivera-Colon  
Institute of Ecology and Evolution  
University of Oregon

